From 367efccd994ef0076ff665770f86ba3321c3efd9 Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Thu, 26 Sep 2024 12:14:14 +0300
Subject: [PATCH] [FSW-9836] fs1: onix: exclude pcie phy fw init

Exclude PCIe PHY FW init from ONIX since PHY FW
memory does not exist in ONIX, and this flow
will cause a crash.

Change-Id: I9c74f502f18a2258302fcf1380b9b18b579142ae
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index 63f45af..5a27150 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -565,7 +565,7 @@
 #define ALIGN_TO_16_BYTES(addr) (((addr) + 15) & ~0xF)
 #define PPBOOT_LAST_ADDR ((uint32_t) (&__SDATA_LOAD))
 
-void hl_init_pcie_phy_fw(void)
+__weak void hl_pcie_phy_fw_copy(void)
 {
 	/* sram_loc_addr located at PPBOOT_LAST_ADDR, aligned to 16 bytes */
 	uint32_t sram_loc_addr = ALIGN_TO_16_BYTES(PPBOOT_LAST_ADDR);
@@ -579,3 +579,8 @@
 	/* Copy CSR_LOC_DATA to IO0_PCIE_FW_CSR_BASE */
 	memcpy((void *)IO0_PCIE_FW_CSR_BASE, (const void *)csr_loc_data, get_csr_locations_data_size());
 }
+
+void hl_init_pcie_phy_fw(void)
+{
+	hl_pcie_phy_fw_copy();
+}
diff --git a/arch/fs1/arc/onix.c b/arch/fs1/arc/onix.c
index e8fc14a..36bfc6d 100644
--- a/arch/fs1/arc/onix.c
+++ b/arch/fs1/arc/onix.c
@@ -17,6 +17,8 @@
 	return SYS_CLOCK_FREQUENCY;
 }
 
+void hl_pcie_phy_fw_copy(void) {}
+
 u32 hl_skip_pcie_init(void)
 {
 	return 1;
